core/domain/state_domain.py

```diff
@@ -65,29 +65,32 @@ def from_dict(cls, answer_group_dict):
             Outcome.from_dict(answer_group_dict['outcome']),
             [RuleSpec.from_dict(rs) for rs in answer_group_dict['rule_specs']],
             answer_group_dict['training_data'],
-            answer_group_dict['tagged_misconception_id']
+            answer_group_dict['tagged_skill_misconception_id']
         )
 
     def __init__(
-            self, outcome, rule_specs, training_data, tagged_misconception_id):
+            self, outcome, rule_specs, training_data,
+            tagged_skill_misconception_id):
         """Initializes a AnswerGroup domain object.
 
         Args:
             outcome: Outcome. The outcome corresponding to the answer group.
             rule_specs: list(RuleSpec). List of rule specifications.
             training_data: list(*). List of answers belonging to training
                 data of this answer group.
-            tagged_misconception_id: int or None. The id of the tagged
-                misconception for the answer group, when a state is part of a
-                Question object that tests a particular skill.
+            tagged_skill_misconception_id: str or None. The format is
+                '<skill_id>-<misconeption_id>'. The skill id of the tagged
```

Comment:
misconception is misspelled.

Reword slightly: The format is XXX, where "skill_id" is YYY and "misconception_id" is ZZZ.

Not sure I understand the "when" clause at the end. Can you make that into a separate sentence and make sure it reads clearly?

Reply:
Done.